test/integration-tests/testset_sdk/javascript/functionProvider.js

```diff
@@ -1,28 +1,31 @@
+// TODO: We need to fix this Jochen
 const iotea = require('../../../../src/module.js');
-//const iotea = require('boschio.iotea');
-
-process.env.MQTT_TOPIC_NS = 'iotea/';
-
-const Logger = iotea.util.Logger;
-process.env.LOG_LEVEL = Logger.ENV_LOG_LEVEL.INFO;
+// const iotea = require('boschio.iotea');
 
 const {
     FunctionTalent,
+    ProtocolGateway
 } = iotea;
 
+const {
+    Logger
+    MqttProtocolAdapter
+} = iotea.util;
+
+process.env.LOG_LEVEL = Logger.ENV_LOG_LEVEL.INFO;
+
 class FunctionProvider extends FunctionTalent {
-    constructor(connectionString) {
-        super('function-provider-js', connectionString);
+    constructor(protocolGatewayConfig) {
+        super('function-provider-js', pro);
 
         // Register Functions
-        this.registerFunction("echo", this.echo.bind(this));
+        this.registerFunction('echo', this.echo.bind(this));
     }
 
     async echo(value, ev, evctx) {
-        this.logger.debug("Echo called");
+        this.logger.debug('Echo called');
         return value;
     }
 }
 
-const t1 = new FunctionProvider('mqtt://localhost:1883');
-t1.start()
+new FunctionProvider(ProtocolGateway.createDefaultConfiguration([ MqttProtocolAdapter.createDefaultConfiguration() ])).start();
```

Comment:
I try to limited the number of functionality (calls) in one line to less as possible. There are many stuff done in only one line. But I agree that the variable itself is not used.

Reply:
I'll resplit